connection.js

```diff
@@ -316,70 +316,20 @@ class Connection {
             return { reject: true, msg: "Currently, " + Object.keys(this.connected_games).length + " games are being played by this bot, maximum is " + config.maxconnectedgames + " (if you see this message and you dont see any game on the bot profile page, it is because private game(s) are being played) , try again later " };
         }
 
-        if ((user.ranking < config.minrank) && !config.minrankranked && !config.minrankunranked) {
-            return minmaxRankFamilyReject("minrank");
-        }
-        if ((user.ranking < config.minrankranked) && notification.ranked) {
-            return minmaxRankFamilyReject("minrankranked");
-        }
-        if ((user.ranking < config.minrankunranked) && !notification.ranked) {
-            return minmaxRankFamilyReject("minrankunranked");
-        }
-        if ((user.ranking > config.maxrank) && !config.maxrankranked && !config.maxrankunranked) {
-            return minmaxRankFamilyReject("maxrank")
-        }
-        if ((user.ranking > config.maxrankranked) && notification.ranked) {
-            return minmaxRankFamilyReject("maxrankranked");
-        }
-        if ((user.ranking > config.maxrankunranked) && !notification.ranked) {
-            return minmaxRankFamilyReject("maxrankunranked");
+        const resultRank = genericMinmaxFamiliesReject("rank", user.ranking, notification.time_control.speed, notification.ranked, notification.handicap, user.username);
+        if (resultRank) {
+            return (resultRank);
```

Comment:
I'm probably at fault for introducing this style, but no need for those brackets here. `return resultRank;` will work too.

Reply:
maybe you're right !
but since i dont know about this i cant tell what should be done except from the safe simple way